convention-plugin-test-option/src/main/kotlin/EmulatorJobsMatrix.kt

```diff
@@ -12,19 +12,63 @@ class EmulatorJobsMatrix {
             .create()
     }
 
-    fun createMatrixJsonFile(rootProject: Project) {
-        val taskList = getTaskList(rootProject = rootProject).map { it.joinToString(separator = " ") }
-        val matrix = mapOf("gradle_tasks" to taskList)
-        val jsonText = gson.toJson(matrix)
-        rootProject.layout.buildDirectory.asFile.get().also { buildDir ->
-            buildDir.mkdirs()
-            File(buildDir, "emulator_jobs_matrix.json").writeText(jsonText)
-        }
+    fun createMatrixJsonFiles(rootProject: Project) {
+        mapOf(
+            "emulator_jobs_matrix.json" to getEmulatorTaskList(rootProject = rootProject),
+            "ios_test_jobs_matrix.json" to getIosTestTaskList(rootProject = rootProject),
+            "js_test_jobs_matrix.json" to getJsTestTaskList(rootProject = rootProject),
+            "jvm_test_jobs_matrix.json" to getJvmTestTaskList(rootProject = rootProject)
+        )
+            .mapValues { entry -> entry.value.map { it.joinToString(separator = " ") } }
+            .forEach { (fileName: String, taskList: List<String>) ->
+                val matrix = mapOf("gradle_tasks" to taskList)
+                val jsonText = gson.toJson(matrix)
+                rootProject.layout.buildDirectory.asFile.get().also { buildDir ->
+                    buildDir.mkdirs()
+                    File(buildDir, fileName).writeText(jsonText)
+                }
+            }
     }
 
-    fun getTaskList(rootProject: Project): List<List<String>> =
+    fun getIosTestTaskList(rootProject: Project): List<List<String>> =
+        rootProject.subprojects.filter { subProject ->
+            listOf(
+                "firebase-analytics"
+            ).contains(subProject.name).not()
+        }.map { subProject ->
+            "${subProject.path}:iosSimulatorArm64Test"
```

Comment:
This should check for the CPU architecture running the gradle script. For X86-64 CPUs it should be using `iosX64Test`